addons/web/static/src/js/fields/abstract_field.js

```diff
@@ -299,6 +299,32 @@ var AbstractField = Widget.extend({
         return this._render() || $.when();
     },
 
+    /**
+     * add the invalid class on a field
+     */
+    setInvalidClass: function () {
```

Comment:
oh, and this method looks a little bit too much 'implementation dependant'.  Maybe it should be renamed to something setInvalidState or something